gp-includes/warnings.php

```diff
@@ -201,49 +213,122 @@ public function warning_length( $original, $translation, $locale ) {
 	 * @param GP_Locale $locale      The locale of the translation.
 	 * @return string|true True if check is OK, otherwise warning message.
 	 */
-	public function warning_tags( $original, $translation, $locale ) {
+	function warning_tags( $original, $translation, $locale ) {
 		$tag_pattern       = '(<[^>]*>)';
 		$tag_re            = "/$tag_pattern/Us";
-		$original_parts    = preg_split( $tag_re, $original, - 1, PREG_SPLIT_DELIM_CAPTURE );
-		$translation_parts = preg_split( $tag_re, $translation, - 1, PREG_SPLIT_DELIM_CAPTURE );
+		$original_parts    = array();
+		$translation_parts = array();
+
+		if ( preg_match_all( $tag_re, $original, $m ) ) {
+			$original_parts = $m[1];
+		}
+		if ( preg_match_all( $tag_re, $translation, $m ) ) {
+			$translation_parts = $m[1];
+		}
+
+		// East asian languages can remove emphasis/italic tags
+		if ( count( $original_parts ) > count( $translation_parts ) ) {
+
+			$languages_without_italics = array(
+				'ja',
+				'ko',
+				'zh',
+				'zh-hk',
+				'zh-cn',
+				'zh-sg',
+				'zh-tw',
+			);
```

Comment:
Can we move this to a class property like `$length_exclude_languages`?

Reply:
Done.